web_poet/page_inputs/client.py

```diff
@@ -39,8 +39,17 @@ class HttpClient:
     :ref:`advanced-downloader-impl` documentation.
     """
 
-    def __init__(self, request_downloader: Callable = None):
+    def __init__(
+        self,
+        request_downloader: Callable = None,
+        *,
+        save_responses: bool = False,
```

Comment:
> I don't currently see how can we set the first flag when we generate a test, maybe it shouldn't be a flag of __init__ but some other check.

So, HttpClient is provided here: https://github.com/scrapinghub/scrapy-poet/blob/cb79d3b870d3cb2cd3f9770e828481a0a51401b3/scrapy_poet/page_input_providers.py#L246

To change an argument for HttpClient, the provider must know if we're in the test generation phase or in the test running / regular execution phase. It seems this can be done through crawler - either through a custom setting (something like `_SCRAPY_POET_SAVEFIXTURE=True`), or by setting some crawler attribute (probably worse, but not sure).